inputs.postgresql fails with panic after postgres12 upgrade

### Relevant telegraf.conf:

```toml
# Read metrics from one or many postgresql servers
[[inputs.postgresql]]
  ## specify address via a url matching:
  ##   postgres://[pqgotest[:password]]@localhost[/dbname]\
  ##       ?sslmode=[disable|verify-ca|verify-full]
  ## or a simple string:
  ##   host=localhost user=pqotest password=... sslmode=... dbname=app_production
  ##
  ## All connection parameters are optional.
  ##
  ## Without the dbname parameter, the driver will default to a database
  ## with the same name as the user. This dbname is just for instantiating a
  ## connection with the server and doesn't restrict the databases we are trying
  ## to grab metrics for.
  ##
  address = "host=HOST user=telegraf password=PASSWORD sslmode=disable dbname=DATABASE"
  ## A custom name for the database that will be used as the "server" tag in the
  ## measurement output. If not specified, a default one generated from
  ## the connection address is used.
  # outputaddress = "db01"

  ## connection configuration.
  ## maxlifetime - specify the maximum lifetime of a connection.
  ## default is forever (0s)
  max_lifetime = "0s"

  ## A  list of databases to explicitly ignore.  If not specified, metrics for all
  ## databases are gathered.  Do NOT use with the 'databases' option.
  # ignored_databases = ["postgres", "template0", "template1"]

  ## A list of databases to pull metrics about. If not specified, metrics for all
  ## databases are gathered.  Do NOT use with the 'ignored_databases' option.
  # databases = ["app_production", "testing"]
```


### System info:

archlinux telegraf-bin 1.12.5 x86_64

### Steps to reproduce:

Same config as for postgres 11 fails now for postgres 12.1

```
panic: interface conversion: interface {} is nil, not string
goroutine 1883 [running]:
github.com/influxdata/telegraf/plugins/inputs/postgresql.(*Postgresql).accRow(0xc000552380, 0x2befb80, 0xc00014f480, 0x2c5abc0, 0xc0001d17a0, 0xc00022dce0, 0x15, 0x15, 0xc00014f480, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/postgresql/postgresql.go:158 +0x7cc
github.com/influxdata/telegraf/plugins/inputs/postgresql.(*Postgresql).Gather(0xc000552380, 0x2c5abc0, 0xc0001d17a0, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/postgresql/postgresql.go:100 +0x1a6
github.com/influxdata/telegraf/internal/models.(*RunningInput).Gather(0xc0001f6140, 0x2c5abc0, 0xc0001d17a0, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/internal/models/running_input.go:110 +0x6d
github.com/influxdata/telegraf/agent.(*Agent).gatherOnce.func1(0xc000b7c3c0, 0xc0001f6140, 0x2c5abc0, 0xc0001d17a0)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:332 +0x3f
created by github.com/influxdata/telegraf/agent.(*Agent).gatherOnce
        /go/src/github.com/influxdata/telegraf/agent/agent.go:331 +0xdc
```


### Expected behavior:

works as with postgres 11

### Actual behavior:

fails with panic

--- a/plugins/inputs/postgresql/postgresql.go
+++ b/plugins/inputs/postgresql/postgresql.go
@@ -155,7 +155,12 @@ func (p *Postgresql) accRow(row scanner, acc telegraf.Accumulator, columns []str
 	}
 	if columnMap["datname"] != nil {
 		// extract the database name from the column map
-		dbname.WriteString((*columnMap["datname"]).(string))
+		if dbNameStr, ok := (*columnMap["datname"]).(string); ok {
+			dbname.WriteString(dbNameStr)
+		} else {
+			// PG 12 adds tracking of global objects to pg_stat_database
+			dbname.WriteString("postgres_global")
+		}
 	} else {
 		dbname.WriteString("postgres")
 	}

